Experimental STT Voice Transcription in Chat

hassoncs opened the pull request:
Add STT or speech-to-text support to the chat input box based on this spec:
https://github.com/Kilo-Org/handbook/pull/99

It uses the OpenAI Whisper API along with FFmpeg in order to transcribe the user's speech to text
It has a proper volume visualizer that shows the amplitude of the input
It's currently behind an experiment and will only be available if users have FFmpeg installed and an OpenAI key set in one of their providers. 

![2025-12-11 10 13 48](https://github.com/user-attachments/assets/3f49e6bc-fed6-44ee-82d3-d9c41740372e)
<img width="382" height="145" alt="CleanShot 2025-12-11 at 10 15 37" src="https://github.com/user-attachments/assets/c0c0f5a7-b92c-47ef-a573-2b672a8c3563" />

jrf0110 commented on src/core/webview/ClineProvider.ts:
```diff
@@ -2204,6 +2204,13 @@ ${prompt}
 
+		// kilocode_change start - checkSpeechToTextAvailable
+		console.log("🎙️ [ClineProvider] Checking speech-to-text availability for webview state update...")
+		const { checkSpeechToTextAvailable } = await import("./speechToTextCheck")
```
Does this need to be a dynamic import?

hassoncs committed:
'chore: WIP'

hassoncs committed:
fix(chat): remove unnecessary focus state and fix recording border styling

- Remove unused isFocused state management from ChatTextArea
- Fix recording border to use inline style with proper error foreground color
- Simplify border logic by removing redundant focus-based styling
- Update Storybook mocks with missing extension state properties
- Add currentApiConfigName and listApiConfigMeta to ChatView stories

hassoncs committed:
chore: add kilocode change markers and clean up unused speech types

- Add // kilocode_change comments to track custom modifications in type definitions
- Remove deprecated speech recognition message types from WebviewMessage interface
- Add descriptive comments to STT-related component imports
- Mark experiment IDs and message types with change tracking comments

hassoncs committed:
chore(changeset): add changeset file for version tracking

hassoncs committed:
Add docs

hassoncs committed:
Update i18n strings

hassoncs committed:
bot fixes

hassoncs committed:
refactor(stt): consolidate FFmpeg detection with path caching

Move FFmpeg availability checking from sttConfig.ts into FFmpegCaptureService
as a static method with global path caching and absolute path resolution.

hassoncs committed:
fix(stt): cache FFmpeg not-found result to avoid repeated checks

Set cachedFFmpegPath to null when FFmpeg is not found to prevent
repeated filesystem path checks on subsequent availability queries.
